lib/Doctrine/ORM/AbstractQuery.php

```diff
@@ -323,7 +325,7 @@ public function getParameters()
      *
      * @param mixed $key The key (index or name) of the bound parameter.
      *
-     * @return Query\Parameter|null The value of the bound parameter, or NULL if not available.
+     * @return Parameter|false|null The value of the bound parameter, or NULL if not available.
```

Comment:
I don't think `false` can actually be returned here: if `$filteredParmeters` is empty, we return `null`

Reply:
You're right. I guess we could make Psalm understand this somehow to solve this on future levels